app/models/services.py

```diff
@@ -12,67 +15,123 @@ class ErrInvalidParameters(Exception):
     pass
 
 
-# DB initializtion
-db = tinydb.TinyDB("db.json")
-services = db.table("services")
-
-
-def insert(shop_id="", name="", description="", images=None):
-    if not images:
-        images = []
-
-    # Initialize id with randomly generated string
-    service_id = util.random_id(initial="s")
-
-    services.insert(
-        {
-            "service_id": service_id,
-            "shop_id": shop_id,
-            "name": name,
-            "slug": slugify(name),
-            "description": description,
-            "images": images,
-        }
-    )
-
-    # TODO: upload images into disk
-    pass
-
-    return service_id
-
-
-def delete(service_id):
-    query = tinydb.Query()
-    services.remove(query.service_id == service_id)
-
-
-def all():
-    results = services.all()
-
-    if not results:
-        raise ErrNotFound("oops.. no service found in the db")
-
-    return results
-
-
-def get(service_id):
-    service = tinydb.Query()
-    result = services.search(service.service_id == service_id)
-
-    if not result:
-        raise ErrNotFound(f"no service found for the provided id: {service_id}")
-
-    return result[0]
-
-
-def get_by_name(name=""):
-    if not name:
-        raise ErrInvalidParameters("service name must be provided")
-
-    service = tinydb.Query()
-    result = services.search(service.name == name)
-
-    if not result:
-        raise ErrNotFound(f"no service found for the provided name: {name}")
-
-    return result[0]
+class ServiceRepo:
```

Comment:
hmmm.. interesting, why the name of this class is `ServiceRepo`? I actually prefer `Service`, is there a reason for Repo?

Reply:
There is also a `Service` class in the same file. So me idea is `Service` class is for handling business logic only like on the other hand a Repo is only know about saving, retrieving data from the database. We can combine those classes into one but the Service is coupled to the database layer.
What do you think? I'm ok if you prefer one class only :)